src/pool/cli.c

```diff
@@ -422,6 +422,13 @@ pool_connect_cp(tse_task_t *task, void *data)
 		D_GOTO(out, rc);
 	}
 
+	rc = dc_mgmt_pool_connect(pool);
+	if (rc != 0) {
+		D_ERROR("failed to register pool connect with agent: "DF_RC"\n",
+			DP_RC(rc));
+		D_GOTO(out, rc);
+	}
```

Comment:
Hm, do we want it to be fatal when we fail to start monitoring the pool handle? I feel like this is a secondary function - an error to log but allow the user to get their pool handle and use it.

Reply:
Unfortunately if we don't clean up these handles it will leak resources server side and also make it so you can't destroy the pool without either forcing the destroy or restarting DAOS.  With some of the other tasks involved I think this feature is more necessary than a nice to have. Although we don't fail if the disconnect isn't possible.

Reply:
Hm, okay. Yeah, that does seem like a problem if there's no way to force it. Though since admins are the ones who destroy pools, it seems like we could let them force it.